eng/azpbuild.ps1

```diff
@@ -0,0 +1,40 @@
+param(
+    [string]$Configuration = "Release",
+    [Parameter(ValueFromRemainingArguments = $true)][String[]]$OtherArgs
+)
+
+$repoRoot = Split-Path $PSScriptRoot -Parent
+$artifactDir = Join-Path $repoRoot "artifacts"
+$tmpDir = Join-Path $artifactDir "tmp"
+$loggerRsp = Join-Path $tmpDir "logger.rsp"
+$loggersZip = Join-Path $tmpDir "msbuildlogger.zip"
+
+if (!(Test-Path $tmpDir)) {
+    mkdir $tmpDir | Out-Null
+}
+
+# Until we figure out a better way to get this...
+# This the same URL used by the Azure tasks: https://github.com/Microsoft/azure-pipelines-tasks/blob/da57c7efbf858a628786c869d7d14b887d1d90d7/Tasks/Common/MSBuildHelpers/make.json#L5
+# And it's HTTPS, so I think we can trust it.
+$loggerUrl = "https://vstsagenttools.blob.core.windows.net/tools/msbuildlogger/3/msbuildlogger.zip"
+Invoke-WebRequest $loggerUrl -OutFile $loggersZip
+Expand-Archive $loggersZip -DestinationPath $tmpDir -Force
```

Comment:
Before removing the "hacky" from the title and undrafting this, I want to see how people feel about this. (@Eilon @pakrym @dougbu @ryanbrandenburg specifically)

We need to get `Microsoft.TeamFoundation.DistributedTask.MSBuild.Logger.dll` on the machine. The URL we're using is what is used when AzP builds their own tasks, but I believe in that case the DLL lives in the Task "bundle" that gets deployed to the agent rather than being downloaded on the fly. There are a few issues with how we're doing this:

1. We don't control this blob share. The file could move or be replaced. It is owned by AzP and accessed via HTTPS, but they could just decide to move it.
1. Since we're downloading a remote file on build, there's a very small possibility for new flakiness here if there's a network disruption. Since we also need NuGet which is on Azure Blobs, I'm not too worried about this.

I do think we should follow up with AzDO about making this logger more readily available so we don't have to do this hack.

Reply:
I say let's continue using this blob just don't fail the build if a download fails.

Reply:
That should be feasible and would reduce the flakiness. We could also embed a hash and do a hash check.

Reply:
@pakrym 's idea seems to be a reasonable strategy for now.

Reply:
> We could also embed a hash and do a hash check.

Why?

Reply:
One possible other option: include a dummy `MSBuild` task invocation in the build YAML, causing Pipelines to make the task (and its copy of the DLL) available on the machine. Then find and use it from there (no idea how to do this).

This might also work only for Windows.

On balance, I prefer your approach.

@chcosta do you have a Pipelines contact we can run this by?

Reply:
@vtbassmat, thoughts?

Reply:
@vtbassmatt thoughts?

Reply:
To summarize: We use a custom PowerShell script to launch MSBuild but still want all the `##vso` logging goop that comes from the logger the official MSBuild task uses. However, we need to get that logger on the box somehow in order to use it. Right now, I'm just getting it from the URL listed in https://github.com/Microsoft/azure-pipelines-tasks/blob/da57c7efbf858a628786c869d7d14b887d1d90d7/Tasks/Common/MSBuildHelpers/make.json#L5 during the build script itself but that seems hacky

Reply:
Grabbing the blob from the network is the better option. Hitting an HTTPS endpoint is the same from everywhere; different agent builds could have subtly different behavior about where they lay out the task cache and so on.

The non-hacky answer is to resolve the question of making the logger available as a NuGet package. I'll ping that thread again.